src/com/sheepit/client/standalone/swing/activity/Working.java

```diff
@@ -454,11 +471,39 @@ public void actionPerformed(ActionEvent e) {
 			Client client = parent.getClient();
 			if (client != null) {
 				if (client.isRunning()) {
-					exitAfterFrame.setText("Cancel exit");
-					client.askForStop();
+					String[] exitJobOptions = {"Finish Current Jobs and then Exit", "Cancel Everything and Exit Now", "Do Nothing"};
```

Comment:
You can probably make these popups less overwhelming by 
a) wrapping the main "Please be aware that" text
b) Making these buttons a bit shorter:
```suggestion
					String[] exitJobOptions = {"Exit after current Jobs", "Exit Immediately", "Do Nothing"};
```